### ISSUE TITLE
Build blog/:article route

### ISSUE THREAD

**Issue by zeim839** [ufosc/Club_Website_2#88]
TODO: Implement the `/blog/:id` route in `app.js` (it currently appears as `/blog/:article` on line 67).  

The server should read the 'id' URL parameter in `req.params.id` and query the database for the corresponding article. If the article ID does not exist, the user should be redirected back to `/blog` and an alert banner should appear to notify the user that the article hasnt been found. You can add an alert banner by adding the following line to `views/article.ejs` (should be the first element below the <body> tag):
```ejs
 <%- include('./components/alert.ejs') %>
```
You can then display an optional error by adding an 'alert: "message"' key to `res.render` (more on that later).

If the article exists and has 'draft' status, then the user will have to be logged in to access the article - otherwise they will be redirected to the blog page as above. If the article is published, then run res.render() with the appropriate keys and database responses, e.g:
```js
return res.render('article', {
  title: TITLE_FROM_DATABASE,
  content: CONTENT_FROM_DATABASE,
  // ...
})
```

**Comment by adrianmarinwork** [comment]
Do you want me to take a look?

**Comment by zeim839** [comment]
@adrianmarinwork 
Yes, I'll assign you to the issue.

However, instead of implementing a new route, lets add this to the`/api/blog/:id` route (it's located in routes/blog.js). Instead of returning article data, the route should check if article ID exists and should then render the article page (see the res.render() calls in app.js). Articles that aren't in the `published` state should only be visible to logged in users.

**Comment by adrianmarinwork** [comment]
Hello, I think I have achieved the objective of this issue. But I think that something is odd from another part of the app (or probably I'm missing something).
The parts of:
- Check if the article existed
- If the user is not logged in and the article is not published then don't show the article
Were already done.
The thing it looks odd is that right now as we are rendering the template from the `/api/blog/:id` the styles are not loading.
Do you know something?
![Captura](https://user-images.githubusercontent.com/118568289/203868796-9113a69d-8b5d-454d-91d2-3722e223adf3.PNG)

**Comment by zeim839** [comment]
I think it's because all the HREF's are relative so they're pointing to the incorrect URL. Can you try removing the `..` before each link?
```ejs
        <title><%= title %></title>
        <link rel="icon" type="image/x-icon" href="../assets/favicon.png" />
        <link rel="stylesheet" href="../css/common.css" />
        <link rel="stylesheet" href="../css/article.css" />
        <script src="../js/navbar.js" defer async></script>
```

**Comment by adrianmarinwork** [comment]
Looks like it works. I have changed the routes of the script tak and the assets, do you want me to keep them like that or only the css files?

![image](https://user-images.githubusercontent.com/118568289/203870489-965dce46-1ffd-43aa-b54d-135e435963b8.png)
![image](https://user-images.githubusercontent.com/118568289/203870527-f4c749f4-23d8-446c-aea3-108607f29b0c.png)

**Comment by zeim839** [comment]
You can keep everything as it is now, it looks like it's working

**Comment by adrianmarinwork** [comment]
Perfect, I'll create the pull request, if you see that somenthing it's not of your like or you prefer to change it go ahead.

**Comment by zeim839** [comment]
Implemented in #98 
Thanks @adrianmarinwork
